src/shared/migrations.ts

```diff
@@ -0,0 +1,19 @@
+import { migrate } from 'postgres-migrations'
+import { Client } from 'pg'
+
+export async function applyMigrations(): Promise<void> {
+  console.log('Applying migrations')
+
+  const dbConfig = {
+    connectionString: process.env.DATABASE_URL
+  }
+
+  const client = new Client(dbConfig)
+  try {
+    await client.connect()
+    await migrate({ client }, './db/migrations')
+  } finally {
```

Comment:
This is missing a catch block

Reply:
It's by design I think. If the migration fails HBP will fail (and restart) and try to apply the migrations again.

Docker should handle the restart with ex [`restart: always`](https://github.com/nhost/hasura-backend-plus/blob/master/examples/simple-hasura-minio/docker-compose.yaml#L26) in docker-compose.

Reply:
yeah, we want the exception to be thrown and the container fail to start.